src/api/Controllers/ExportController.cs

```diff
@@ -182,7 +189,65 @@ public async Task<IActionResult> ExportCsvAsync([FromQuery][MaxLength(MaxPageSiz
-        return File(Encoding.UTF8.GetBytes(stringWriter.ToString()), "text/csv", "boreholes_export.csv");
+        return File(Encoding.UTF8.GetBytes(stringWriter.ToString()), "text/csv", $"{ExportFileName}_{DateTime.UtcNow:yyyyMMddHHmmss}.csv");
+    }
+
+    /// <summary>
+    /// Asynchronously gets all <see cref="Borehole"/> with attachments filtered by ids.
+    /// </summary>
+    /// <param name="ids">The required list of borehole ids to filter by.</param>
+    /// <returns>A ZIP file including the borehole data as JSON and all corresponding attachments.</returns>
+    [HttpGet("zip")]
+    [Authorize(Policy = PolicyNames.Viewer)]
+    public async Task<ActionResult> ExportJsonWithAttachmentsAsync([FromQuery][MaxLength(MaxPageSize)] IEnumerable<int> ids)
+    {
+        var idList = ids?.ToList() ?? [];
+        if (idList.Count < 1)
+        {
+            return BadRequest("The list of IDs must not be empty.");
+        }
+
+        try
+        {
+            var boreholes = await context.Boreholes.GetAllWithIncludes().AsNoTracking().Where(borehole => idList.Contains(borehole.Id)).ToListAsync().ConfigureAwait(false);
+            var files = await context.BoreholeFiles.Include(f => f.File).AsNoTracking().Where(f => idList.Contains(f.BoreholeId)).ToListAsync().ConfigureAwait(false);
+            var fileName = $"{ExportFileName}_{DateTime.UtcNow:yyyyMMddHHmmss}";
+
+            // If only one borehole is exported, use its name as the file name
+            if (idList.Count == 1)
+            {
+                fileName = boreholes.Single().Name;
+            }
+
+            var json = JsonSerializer.Serialize(boreholes, jsonExportOptions);
+
+            using var memoryStream = new MemoryStream();
+            using (var archive = new ZipArchive(memoryStream, ZipArchiveMode.Create, true))
+            {
+                // Add JSON file with borehole data
+                var jsonEntry = archive.CreateEntry($"{fileName}.json", CompressionLevel.Fastest);
+                using var entryStream = jsonEntry.Open();
+                using (var textWriter = new StreamWriter(entryStream))
+                {
+                   await textWriter.WriteAsync(json).ConfigureAwait(false);
+                }
+
+                foreach (var file in files)
+                {
+                    var fileBytes = await boreholeFileCloudService.GetObject(file.File.NameUuid!).ConfigureAwait(false);
+                    var zipEntry = archive.CreateEntry($"{file.BoreholeId}_{file.File.Name}", CompressionLevel.Fastest);
+                    using var zipEntryStream = zipEntry.Open();
+                    await zipEntryStream.WriteAsync(fileBytes.AsMemory(0, fileBytes.Length)).ConfigureAwait(false);
+                }
+            }
+
+            return File(memoryStream.ToArray(), "application/zip", $"{fileName}");
+        }
+        catch (Exception ex)
+        {
+            logger.LogError("Failed to prepare ZIP file: {Exception}.", ex);
```

Comment:
REQ: Auf diese Weise wird `ex.ToString()` aufgerufen und _nur_ die Exception-Message geloggt. Im Log möchten wir jedoch die gesamten Infos mit dem StackTrace haben, deshalb wäre folgender Aufruf vermutlich der korrekte:
`logger.LogError(ex, "Failed to prepare ZIP file.", ex);`